planner/optimize.go

```diff
@@ -264,6 +293,22 @@ func optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is in
 		return nil, nil, 0, err
 	}
 
+	if !sctx.GetSessionVars().InRestrictedSQL {
```

Comment:
enclose it as a func？

Reply:
This variable is referenced directly in many places in the codebase, so I don't think we need a function call here.